kgl-math/build.gradle.kts

```diff
@@ -0,0 +1,62 @@
+import codegen.math.GenerateMath
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTargetPreset
+
+plugins {
+	kotlin("multiplatform")
+	`maven-publish`
+}
+
+val generateMath by tasks.registering(GenerateMath::class) {
```

Comment:
Will have to add a dependency on this, to the `KotlinCompile` task. I'm not confident the CI is.... oof, I just realised CI isn't running on this PR. :grimacing:  I'll fix that in a bit.

Reply:
I believe it's an implicit dependency because of the `kotlin.srcDir(generateMath.map { it.commonDir })` in the common source set definition, but it's not bad to make it explicit.

Reply:
Adding `tasks.withType<KotlinCompile> { dependsOn(generateMath) }` throws NullPointerException when running build... not sure what that's about

Reply:
Rip

Reply:
I'll add some tests to see if it's even necessary to explicitly depend on the tasks, since the generation of sources happens before the compilation already, and the source set is picking up the generated sources.